src/state.rs

```diff
@@ -689,6 +697,30 @@ mod tests {
     use super::*;
     use crate::ProgressBar;
 
+    impl BarState {
+        /// Custom constructor for testing duration on finish
+        fn with_start_at(start_at: Instant) -> Self {
+            BarState {
+                draw_target: ProgressDrawTarget::hidden(),
+                on_finish: ProgressFinish::default(),
+                style: ProgressStyle::default_bar(),
+                state: {
+                    ProgressState {
+                        pos: Arc::new(AtomicPosition::new()),
+                        len: None,
+                        tick: 0,
+                        status: Status::InProgress,
+                        started: start_at,
+                        est: Estimator::new(start_at),
+                        message: TabExpandedString::NoTabs("".into()),
+                        prefix: TabExpandedString::NoTabs("".into()),
+                    }
+                },
+                tab_width: DEFAULT_TAB_WIDTH,
+            }
+        }
+    }
```

Comment:
This seems pretty unidiomatic. Please turn it into a `state_started_at()` helper function that is defined just below the tests that use it.